### PR TITLE
[MFTF] Add product to the Cart if Backorders are allowed on Product level and Product Qty<=0

### LINKED ISSUE #33635: [Issue] [MFTF] Add product to the Cart if Backorders are allowed on Product level and Product Qty<=0
This issue is automatically created based on existing pull request: magento/magento2#33431: [MFTF] Add product to the Cart if Backorders are allowed on Product level and Product Qty<=0

---------
### Description (*)
This is to check if Product can be added to the Cart if Backorders are allowed on Product level and Product Qty<=0

### Manual testing scenarios (*)
1. Login as Admin
2. Open the Edit Product page->Advanced Inventory
3. Set the "Backorders" select value to "Allow Qty Below 0"
4. Set product Qty to <0
5. Save product
6. Open the Product on the Storefront
7. Click the "Add to Cart" button
8. Assert that the Product is in the Cart

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
This is to check if Product can be added to the Cart if Backorders are allowed on Product level and Product Qty<=0

### Manual testing scenarios (*)
1. Login as Admin
2. Open the Edit Product page->Advanced Inventory
3. Set the "Backorders" select value to "Allow Qty Below 0"
4. Set product Qty to <0
5. Save product
6. Open the Product on the Storefront
7. Click the "Add to Cart" button
8. Assert that the Product is in the Cart

### Resolved issues:
1. [x] resolves magento/magento2#33635: [MFTF] Add product to the Cart if Backorders are allowed on Product level and Product Qty<=0

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run all tests

**Comment by AnnaAPak** [comment]
@magento run all tests

**Comment by shikhamis11** [comment]
@magento run Unit Tests, Functional Tests EE

**Review by Den4ik — APPROVED** [review]
✅ Approved.
Looks good for me

**Comment by Den4ik** [comment]
@magento run Functional Tests EE

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9150](https://jira.corp.magento.com/browse/ENGCOM-9150) has been created to process this Pull Request

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-9150](https://jira.corp.magento.com/browse/ENGCOM-9150) has been created to process this Pull Request

File: app/code/Magento/Catalog/Test/Mftf/Test/StorefrontAddProductWithBackordersAllowedOnProductLevelToCartTest.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            <createData entity="SimpleProductInStockQuantityZero" stepKey="createProduct"/>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+        </before>
+
+        <after>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+            <deleteData createDataKey="createProduct" stepKey="deleteProduct"/>
+        </after>
+
+        <actionGroup ref="NavigateToCreatedProductEditPageActionGroup" stepKey="openEditBundleProduct">
```
Comment (shikhamis11) [inline app/code/Magento/Catalog/Test/Mftf/Test/StorefrontAddProductWithBackordersAllowedOnProductLevelToCartTest.xml]: @AnnaAPak Thank you for the pull request.
It looks good. 

I would like you to change the `stepKey` to more relevant. You can make it `openEditCreatedProduct` or `openEditSimpleProduct`

**Review by shikhamis11 — CHANGES_REQUESTED** [review]
https://github.com/magento/magento2/pull/33431/files#r671635660

**Comment by AnnaAPak** [comment]
@magento run all tests

**Comment by AnnaAPak** [comment]
@shikhamis11 requested changes have been added. Can you please review? Thanks.

**Comment by shikhamis11** [comment]
@magento run Unit Tests, Functional Tests EE, Functional Tests B2B

**Comment by shikhamis11** [comment]
@magento run Functional Tests B2B

**Review by shikhamis11 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, thank you for the review.
[ENGCOM-9150](https://jira.corp.magento.com/browse/ENGCOM-9150) has been created to process this Pull Request

**Comment by engcom-Charlie** [comment]
@magento create issue

**Comment by engcom-Charlie** [comment]
:heavy_check_mark: QA Passed

The newly created auto tests are valid.

The test case is working fine. Please find the Allure report for the same.

image

**Pull request merged**